config(course): add title update license fields

HussainTaj-arbisoft opened the pull request:
### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4065

### Description (What does it do?)

This PR adds a help text to the title. The title is a required field and is not required to be defined, but it has to be defined to add the help text.

This PR also adds a new license option `BY CC-SA` and changes default for external resources as requested in https://github.com/mitodl/hq/issues/4065.

### Screenshots (if appropriate):

<img width="544" alt="Screenshot 2024-04-24 at 3 29 48 PM" src="https://github.com/mitodl/ocw-hugo-projects/assets/71316217/27ede97a-2965-41a1-8bad-a36dc5ae772c">


### How can this be tested?

1. Checkout branch `hussaintaj/external-resource-title-help-text` in your local clone of `ocw-hugo-projects`.
2. Navigate to your local `ocw-studio` project.
3. Replace the contents of `ocw-studio/localdev/configs/default-course-config.yml` with the content of `ocw-hugo-projects/ocw-course-v2/ocw-studio.yaml`.
4. Run
    ```sh
    docker compose exec web ./manage.py override_site_config -c localdev/configs/default-course-config.yml -s ocw-course-v2
    ```
5. Open [Studio](http://localhost:8043).
6. Open a course.
7. Open the `External Resources` tab.
8. Add a new external resource.
9. Expect to see the help text for the title field.
10. Expect to see a new license option `BY CC-SA`.
11. Except the default value to be `All Rights Reserved`.
12. Fill in the form and save the external resource.
13. Open/Create a page.
14. Add a link to the external resource via the `Add Link` button in the CKEditor UI.
15. Save the page.
16. Expect the previous step to succeed.

pdpinch commented:
This looks good to me. Note that there is another request for changes to these field though: https://github.com/mitodl/hq/issues/4065

I don't know if one PR is easier than two, but I expect that putting them into the same release would be simpler.

HussainTaj-arbisoft commented:
> I don't know if one PR is easier than two, but I expect that putting them into the same release would be simpler.

Should be simple enough to add it to this PR, seems okay since we need it by tomorrow.

> I don't think that this PR should be modifying `RELEASE.rst` and `VERSION`; that's probably an unintentional change.

pre-commit ci added those changes. RELEASE.rst seems to be a one-time change (adds a line break at the end). We might need to exclude VERSION in the pre-commit config.

HussainTaj-arbisoft commented:
@pdpinch, I've added the license options as requested in https://github.com/mitodl/hq/issues/4065. I've also added `CC BY-SA` to the resource's license field, let me know if that wasn't desired.

@pt2302, I've excluded the files you mentioned from pre-commit. The pre-commit check is failing, probably because of an unrelated reason. I'll ask DevOps to look at it.

HussainTaj-arbisoft commented:
@pdpinch, do the changes seem okay to you too?

pdpinch reviewed (approved):
👍 

you can take or leave my comment
